src/Traces.Core/Services/Traces/TraceService.cs

```diff
@@ -140,8 +140,14 @@ public async Task<TraceDto> CreateTraceAsync(CreateTraceDto createTraceDto)
 
             var traceFiles = new List<TraceFile>();
             var filesToUpload = createTraceDto.FilesToUpload.ValueOr(new List<CreateTraceFileDto>());
+
             if (filesToUpload.Any())
             {
+                if (!createTraceDto.FileContainsNoPii)
+                {
+                    throw new BusinessValidationException(TextConstants.ConfirmFileContainsNoPiiErrorMessage.Trim('"'));
```

Comment:
```suggestion
                    throw new BusinessValidationException(TextConstants.ConfirmFileContainsNoPiiErrorMessage);
```

No need for the trim